lti_consumer/plugin/views.py

```diff
@@ -416,3 +425,196 @@ def scores(self, request, *args, **kwargs):  # pylint: disable=unused-argument
+        """
+        Parse role filter from query param and map it to edx role.
+        """
+        if nrps_filter_role:
+            for edx_role, context_roles in LTI_1P3_CONTEXT_ROLE_MAP.items():
+                if nrps_filter_role in context_roles:
+                    return edx_role
+
+        # if no NRPS filter role provided or there was no match in LTI_1P3_CONTEXT_ROLE_MAP
+        return None
+
+    @staticmethod
+    def build_filter_kwargs(filter_by=None):
+        """
+        Helper method to build filter arguments for get_course_membership python API
+        Returns:
+            access_roles (list): A list of roles if there any, otherwise None
+            include_students (bool)
+        """
+        access_roles = None
+        include_students = True
+
+        if filter_by is None:
+            # if no role filter given, we are interested in instructor & staff. other access
+            # roles like beta tester etc doesn't have equivalent role in LTI Context Role vocab.
+            access_roles = ['instructor', 'staff']
+        elif filter_by != 'student':
+            access_roles = [filter_by]
+            # exclude students if filtered by any access role
+            include_students = False
+
+        return access_roles, include_students
+
+    def preprocess(self, data):
+        """
+        Preprocess the output of get_membership method. It now does followings -
+            - Append external ids to the user
+            - Makes user profile picture link absolute
+        """
+        processed_data = []
+
+        # batch get or create external ids for all users
+        user_ids = [user['id'] for user in data]
+        users = get_user_model().objects.filter(id__in=user_ids)
+        external_ids = compat.batch_get_or_create_externalids(users)
+
+        for user in data:
+            # append external ids to user dictonary
+            user['external_id'] = external_ids[user['id']].external_user_id
+
+            # if user's image exists turn it to an absolute url
+            if user['profile']['profile_image']['has_image']:
+                picture_url = user['profile']['profile_image']['image_url_small']
+                user['profile']['profile_image']['image_url_small'] = self.request.build_absolute_uri(picture_url)
+            processed_data.append(user)
+
+        return processed_data
+
+    def get_serializer(self, data, **kwargs):  # pylint: disable=arguments-differ
+        """
+        Override get_serializer method to change data structure
+        to match LtiNrpsContextMembershipSerializer.
+        """
+
+        # prepare datastructure supported by serializer
+        result = {
+            'id': self.request.build_absolute_uri(),
+            'context': {
+                'id': self.request.lti_configuration.location.course_key
+            },
+            'members': self.preprocess(data),
+        }
+        return super().get_serializer(result)
```

Comment:
It doesn't seem like this is needed anymore, this is being done on the `list` method.

```suggestion
```

Reply:
Sorry, I forgot to remove that. Removed now.